### ISSUE TITLE
[Regression] Supply: Rollout attempts to roll out retained version

### ISSUE THREAD

**Issue by kschults** [fastlane/fastlane#15687]
### New Regression Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Regression Information

- Breaking version: 2.135.0
- Last working version: 2.134.0

### Regression Description

I am using fastlane to start a rollout, while retaining an old version. Previously, I called:

```
    upload_to_play_store(
        track: 'rollout',
        aab: <filepath>,
        rollout: 0.01,
        version_codes_to_retain: [<N1>],
        skip_upload_apk: 'true',
        skip_upload_images: 'true',
        skip_upload_screenshots: 'true',
        validate_only: 'true'
    )
```

The output was:
```
[12:01:34]: Preparing to upload for language 'en-US'... [repeated for all languages]
[12:01:39]: Preparing aab at path '<filepath>' for upload...
[12:02:43]: Updating changelog for code version '<N2>' and language 'en-US'... [repeated for all languages]
[12:02:46]: Updating track 'rollout'...
[12:02:47]: Validating all changes with Google Play...
[12:02:48]: Successfully validated the upload to Google Play
```

Now, with the removal of the "rollout" track, I call:

```
    upload_to_play_store(
        track: 'production',
        aab: <filepath>,
        rollout: 0.01,
        version_codes_to_retain: [<N1>],
        skip_upload_apk: 'true',
        skip_upload_images: 'true',
        skip_upload_screenshots: 'true',
        validate_only: 'true'
    )
```

And the output is:

```
[12:08:24]: Preparing aab at path '<filepath>' for upload...
[12:09:26]: Updating track 'production'...
[12:09:27]: Updating <N1>'s rollout to '0.01' on track 'production'...
[12:09:28]: Validating all track and release changes with Google Play...
[12:09:30]: Successfully validated the upload to Google Play
```

This seems wrong. From the output, it appears that Fastlane is attempting to roll out my retained version N1 to 1%. I want to retain version N1 at 100% and start a rollout of the newly uploaded file with version N2 at 1%.

It's also worth noting that the changelog language uploads aren't happening anymore - should I file that as a separate issue, or is it just not logging that anymore?

### Environment

<details>
<pre>
<details><summary>🚫 fastlane environment 🚫</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.14.6                                     |
| Ruby                        | 2.4.6                                       |
| Bundler?                    | true                                        |
| Git                         | git version 2.21.0 (Apple Git-122.2)        |
| Installation Source         | ~/.rvm/gems/ruby-2.4.6/bin/fastlane         |
| Host                        | Mac OS X 10.14.6 (18G1012)                  |
| Ruby Lib Dir                | ~/.rvm/rubies/ruby-2.4.6/lib                |
| OpenSSL Version             | OpenSSL 1.1.1c  28 May 2019                 |
| Is contained                | false                                       |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 11.2.1                                      |

### System Locale

| Error                       |
| --------------------------- |
| No Locale with UTF8 found 🚫 |

### fastlane files:

<details><summary>`./fastlane/Appfile`</summary>

```ruby
json_key_file("./google_api_key.json")
package_name("<package>")

```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.136.0 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.1.0        |
| executable-hooks          | 1.6.0        |
| bundler-unload            | 1.0.2        |
| rubygems-bundler          | 1.4.5        |
| bundler                   | 1.17.3       |
| CFPropertyList            | 3.0.1        |
| public_suffix             | 2.0.5        |
| addressable               | 2.7.0        |
| atomos                    | 0.1.3        |
| babosa                    | 1.0.3        |
| claide                    | 1.0.3        |
| colored                   | 1.2          |
| colored2                  | 3.1.2        |
| colorize                  | 0.8.1        |
| highline                  | 1.7.10       |
| commander-fastlane        | 4.4.6        |
| declarative               | 0.0.10       |
| declarative-option        | 0.1.0        |
| diff-lcs                  | 1.3          |
| digest-crc                | 0.4.1        |
| unf_ext                   | 0.0.7.6      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20190701 |
| dotenv                    | 2.7.5        |
| emoji_regex               | 1.0.1        |
| excon                     | 0.68.0       |
| multipart-post            | 2.0.0        |
| faraday                   | 0.17.0       |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| faraday_middleware        | 0.13.1       |
| fastimage                 | 2.1.7        |
| gh_inspector              | 1.1.3        |
| jwt                       | 2.1.0        |
| memoist                   | 0.16.1       |
| multi_json                | 1.14.1       |
| os                        | 1.0.1        |
| signet                    | 0.12.0       |
| googleauth                | 0.6.7        |
| httpclient                | 2.8.3        |
| mime-types-data           | 3.2019.1009  |
| mime-types                | 3.3          |
| uber                      | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 3.1.2        |
| google-api-client         | 0.23.9       |
| google-cloud-env          | 1.3.0        |
| google-cloud-core         | 1.4.1        |
| google-cloud-storage      | 1.16.0       |
| json                      | 2.2.0        |
| mini_magick               | 4.9.5        |
| multi_xml                 | 0.6.0        |
| plist                     | 3.5.0        |
| rubyzip                   | 1.3.0        |
| security                  | 0.1.3        |
| naturally                 | 2.2.0        |
| simctl                    | 1.6.6        |
| slack-notifier            | 2.3.2        |
| terminal-notifier         | 2.0.0        |
| unicode-display_width     | 1.6.0        |
| terminal-table            | 1.8.0        |
| tty-screen                | 0.7.0        |
| tty-cursor                | 0.7.0        |
| tty-spinner               | 0.9.1        |
| word_wrap                 | 1.0.0        |
| nanaimo                   | 0.2.6        |
| xcodeproj                 | 1.13.0       |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.3.0        |
| xcpretty-travis-formatter | 1.0.0        |
| imagesize                 | 0.1.1        |
| mini_portile2             | 2.4.0        |
| nokogiri                  | 1.10.4       |
| rspec-support             | 3.8.2        |
| rspec-core                | 3.8.2        |
| rspec-expectations        | 3.8.4        |
| rspec-mocks               | 3.8.1        |
| rspec                     | 3.8.0        |
| sqlite3                   | 1.4.1        |
</details>

*generated on:* **2019-11-20**
</details>
</pre>
</details>

**Comment by joshdholtz** [comment]
@kschults Hey! I will look into replicating this on my end. Hopefully can get a fix out for you soon 👌

**Comment by kschults** [comment]
Thanks @joshdholtz. Would love if it could get into a release before the v2 api gets deprecated.

**Comment by joshdholtz** [comment]
For sure! I want to get this fixed by tomorrow

**Comment by joshdholtz** [comment]
@kschults This may actually be fixed by the lastest master now that I think about it 🤔 I'm going to make a new release so that version might actually work for you 🙃

**Comment by funkybaboon** [comment]
@joshdholtz I wanted to upload an apk to an internal track, which got uploaded straight into production. Is this the same issue?

```
$ fastlane supply --json_key ${AUTHENTICATION_JSON} --skip_upload_metadata --skip_upload_images --skip_upload_screenshots --track_promote_to internal --package_name ${PACKAGE_NAME} --apk ${PACKAGE_NAME}_*.apk
[21:43:10]: Get started using a Gemfile for fastlane https://docs.fastlane.tools/getting-started/ios/setup/#use-a-gemfile

+-------------------------+---------------------------------------------------------------+
|                               Summary for supply 2.137.0                                |
+-------------------------+---------------------------------------------------------------+
| json_key                | ***                                                           |
| skip_upload_metadata    | true                                                          |
| skip_upload_images      | true                                                          |
| skip_upload_screenshots | true                                                          |
| track_promote_to        | internal                                                      |
| package_name            | ***                                                           |
| apk                     | ***                                                           |
| release_status          | completed                                                     |
| track                   | production                                                    |
| skip_upload_apk         | false                                                         |
| skip_upload_aab         | false                                                         |
| skip_upload_changelogs  | false                                                         |
| validate_only           | false                                                         |
| check_superseded_tracks | false                                                         |
| timeout                 | 300                                                           |
| deactivate_on_promote   | true                                                          |
+-------------------------+---------------------------------------------------------------+

[21:43:11]: Preparing apk at path '**' for upload...
[21:43:18]: 	Version Code: 347
[21:43:18]: Updating track 'production'...
[21:43:18]: Uploading all changes to Google Play...
[21:43:21]: Successfully finished the upload to Google Play
```

**Comment by kschults** [comment]
@funkybaboon I think your issue is that you should use `--track internal` instead of `--track_promote_to internal` (or you might need both). The default track is production, so that's what it's acting on.

**Comment by kschults** [comment]
This seems to be fixed in validation mode in 2.137.0.

**Comment by funkybaboon** [comment]
@kschults Thanks for the hint. It worked with `track_promote_to` for years, so this must be an unintentional change.
